kubernetes/eventhandler.go

```diff
@@ -205,19 +230,31 @@ func (n *nodePodUpdater) OnUpdate(obj interface{}) {
 		return
 	}
 
-	// n.store.List() returns a snapshot at this point. If a delete is received
-	// from the main watcher, this loop may generate an update event after the
-	// delete is processed, leaving configurations that would never be deleted.
-	// Also this loop can miss updates, what could leave outdated configurations.
-	// Avoid these issues by locking the processing of events from the main watcher.
 	if n.locker != nil {
 		n.locker.Lock()
 		defer n.locker.Unlock()
 	}
-	for _, pod := range n.store.List() {
-		pod, ok := pod.(*Pod)
-		if ok && pod.Spec.NodeName == node.Name {
-			n.handler(pod)
+
+	// Slice includes the old and new version of caching object that changes in the current update event. slice[0] is the old version and slice[1] the new updated one
+	slice := n.nodewatcher.Deltaobjects()
+	cachednodeold, ok := slice[0].(*Node)
+
+	if node.Name == cachednodeold.Name && ok {
+		labelscheck := checkMetadata(node.ObjectMeta.Labels, cachednodeold.ObjectMeta.Labels)
+		annotationscheck := checkMetadata(node.ObjectMeta.Annotations, cachednodeold.ObjectMeta.Annotations)
+		// Only if there is a diffrence in Metadata labels or annotations proceed to Pod update
+		if !labelscheck || !annotationscheck {
+			// n.store.List() returns a snapshot at this point. If a delete is received
+			// from the main watcher, this loop may generate an update event after the
+			// delete is processed, leaving configurations that would never be deleted.
+			// Also this loop can miss updates, what could leave outdated configurations.
+			// Avoid these issues by locking the processing of events from the main watcher.
```

Comment:
same as above for moving the comment or the code

Reply:
Actually I now remembered that deliberately moved that comment here just before the n.store.List()

Before it was [here](https://github.com/elastic/elastic-agent-autodiscover/blob/main/kubernetes/eventhandler.go#L160) but I call the Deltaobjects in the middle.

Nevertheless I moved it before lock